### PR TITLE
Add nonce IV

### PR DESCRIPTION
**Pull request by dandydavey**
Encrypting nonce separately for now to keep interfaces tidy, I want to separately think about making versioning easy so we can make further changes (such at concatenating directly with plaintext) lightweight.

**Testing done**
encrypt and decrypt roundtrip works in the privy-crypto tests

### CODE REVIEW

**Review by benjreinhart — COMMENTED** [review]
I think I may have misunderstood our plans. I was assuming we were leaning instead towards encrypting the nonce with the plaintext. That would make our serialization simpler (doesn't append an extra value to the serialized result) and reduce the need for an additional IV. It comes at the cost of backward incompatibilities, but maybe that's ok given how few entries exist in production?

**Comment by dandydavey** [comment]
synced offline, I went with this set of changes because I had it very concrete in my mind that it's a 15min change, whereas combining the nonce raises some scope creep questions re: adding in the bit to indicate whether the nonce is included or not, which would introduce more branching logic etc. / if not adding the bit in, there's some naming questions on how to keep track of when the nonce is part of the ciphertext vs not. 

So I went with the 15 min change and carved out thinking about how to preserve flexibility here so we feel confident making these changes to the library iteratively. But the other quick-and-dirty way of combining the nonce thing is also fine by me if it's similarly a 15 min change, as we talked about offline, not married to this!

**Comment by dandydavey** [comment]
@benjreinhart just resolved the merge conflicts with latest main. Aprooov?

File: src/engines/x0.ts [hunk 1]
```diff
@@ -400,14 +424,18 @@ export class Decryption {
...
       // Decrypt plaintext.
-      const plaintext = await aesGCMDecrypt(this._ciphertext, this._initializationVector, dataKey);
+      const plaintext = await aesGCMDecrypt(
+        this._ciphertext,
+        this._dataInitializationVector,
+        dataKey,
+      );
 
-      // If commitmentId passed in, perform integrity check against the commitment id.
+      // If commitmentHash passed in, perform integrity check against the commitmentHash.
```
Comment (benjreinhart) [inline src/engines/x0.ts:433]: Why this change in wording?

**Review by benjreinhart — APPROVED** [review]

File: src/engines/x0.ts (hunk 1 above)
Comment (dandydavey) [reply src/engines/x0.ts:433]: oops, think this was a merge resolution mistake, but moot point as it's removed in https://github.com/horkos-labs/privy-crypto-js/pull/12!

**Comment by dandydavey** [comment]
Thanks for the review!
